src/protocols/secure_channel/tests/TestCASESession.cpp

```diff
@@ -57,6 +57,20 @@ using TestContext = Test::LoopbackMessagingContext;
 namespace chip {
 namespace {
 
+void ServiceEvents(TestContext & ctx)
+{
+    // Service any messages
+    ctx.DrainAndServiceIO();
+
+    // Messages may have scheduled work, so service them
```

Comment:
DrainAndServiceIO is supposed to handle this.

Stopping and restarting the event loop here is a bit weird.  We should figure out something better.

Reply:
https://github.com/project-chip/connectedhomeip/pull/25104 will address this.